Automattic/woocommerce-payments
# Add jetpack-sync package to support Woo Mobile  (obsolete)

Fixes #3270

**Outstanding questions: p91TBi-6ek-p2#comment-7054**

**Potential issues:** 

- [x] Right after connecting Jetpack on WooCommerce Payments, `woocommerce_is_active` is not switched to `true`. It becomes `true` after several plugin deactivations/activations. p1636120126004500-slack-CBG1CP4EN. This PR adds a new method to hanlde this nicely https://github.com/Automattic/jetpack/pull/21676

- [x] [Test failure for WP 5.4](https://github.com/Automattic/woocommerce-payments/runs/4149870397?check_suite_focus=true) due to `wp_get_environment_type` function had not been added in WP 5.4. p1636450075116700-slack-CBG1CP4EN

- [ ] Ensure that data-sync settings from WCPay does not override ones from other plugins such as Jetpack Backup, etc. This is being handled here https://github.com/Automattic/jetpack/pull/22458

#### Changes proposed in this Pull Request

- Add Jetpack Sync package. 
- Configure Jetpack Sync to two modules: Options and Plugins, which is enough to let `/v1.2/me/sites` works properly for Woo Mobile apps. 









#### Testing instructions

For a dev site, currently, you need to do this: 

- Apply this PR.
- Turn on your jurassic.tube site address: `npm run tube:start`
- Add this line before my current change for woocommerce-payments.php, which helps passing this check `Automattic\Jetpack\Sync\Actions::sync_allowed()`: 

```php
	add_filter( 'jetpack_is_staging_site', '__return_false' );
``` 

or comment this line in wp-config.php, which is set up with Docker:

```php
define( 'WP_ENVIRONMENT_TYPE', 'development' );
``` 
 
- Try updating options in WP Admin > Settings, or activating/deactivating plugins in WP Admin > Plugins. 
- Monitor `options` data for your jurassic.tube site from this endpoint `/v1.2/me/sites`. 
- Expectation 1: `options` data reflects the change made in your jurassic.tube site.
- Expectation 2: `options.woocommerce_is_active` is true when WooCommerce core is active on your jurassic.tube site.

A few internal links: 2a9c9-pb






-------------------


**Post merge**
